### PR TITLE
Some concepts were used before explaining them.

### PR DESCRIPTION
**Pull request by plijnzaad**
Things like function, argument, default were IMHO not explained before actually using them, so I rewrote parts of that.
The concepts object-oriented and functional are not useful for novices

### CODE REVIEW

**Comment by plijnzaad** [comment]
OK folks, very first commit, I hope I haven't offended anyone!

File: lessons/R/00-before-we-start.Rmd [hunk 1]
```diff
@@ -61,7 +61,7 @@ for statistics but also data science. Inspired by the programming language S.
   industry.
 * Available on all platforms.
 * Not just for statistics, but also general purpose programming.
-* Is object oriented and functional.
+* For people who have in experience programmming: R is both an object-oriented and *functional* language
```
Comment (fmichonneau) [inline lessons/R/00-before-we-start.Rmd]: have in experience programmming --> have experience in programming

File: lessons/R/00-before-we-start.Rmd [hunk 2]
```diff
@@ -71,34 +71,41 @@ right of a `#` is ignored by R.
...
 In RStudio, typing `Alt + -` (push `Alt`, the key next to your space bar at the
 same time as the `-` key) will write ` <- ` in a single keystroke.
 
-`=` should only be used to specify the values of arguments in functions for
-instance `read.csv(file="data/some_data.csv")`.
-
 ## Functions and their arguments
 
-Let's look at a simple function call:
+Functions are "canned scripts" that automate doing something complicated or convenient or both.
```
Comment (fmichonneau) [inline lessons/R/00-before-we-start.Rmd]: remove "doing"

File: lessons/R/00-before-we-start.Rmd [hunk 3]
```diff
@@ -71,34 +71,41 @@ right of a `#` is ignored by R.
...
 
-`=` should only be used to specify the values of arguments in functions for
-instance `read.csv(file="data/some_data.csv")`.
-
 ## Functions and their arguments
 
-Let's look at a simple function call:
+Functions are "canned scripts" that automate doing something complicated or convenient or both. 
+Many functions are predefined, or 
+becoming available when using a `library` (more on that later). A function usually gets one or more inputs called *arguments*. Functions often (but not always) return a *value*.
```
Comment (fmichonneau) [inline lessons/R/00-before-we-start.Rmd]: becoming -> become

Comment (fmichonneau) [inline lessons/R/00-before-we-start.Rmd]: when using a `library` --> when using the function `library()`

File: lessons/R/00-before-we-start.Rmd [hunk 4]
```diff
@@ -71,34 +71,41 @@ right of a `#` is ignored by R.
...
 
-Let's look at a simple function call:
+Functions are "canned scripts" that automate doing something complicated or convenient or both. 
+Many functions are predefined, or 
+becoming available when using a `library` (more on that later). A function usually gets one or more inputs called *arguments*. Functions often (but not always) return a *value*. 
+A typical example would be the function `sqrt`. The input (the argument) must be a number, and the return value  (in fact, the output) is the square root of that number. Executing a function ('running it') is called *calling* the function. An example of a function call is:
+
+`b <- sqrt(a)`
+
+Here, the value of `b` is given to the `sqrt`-function, the `sqrt`-function calculates the square root, and returns
```
Comment (fmichonneau) [inline lessons/R/00-before-we-start.Rmd]: `sqrt`-function --> `sqrt()` function

**Comment by fmichonneau** [comment]
the rest looks good! thanks!

**Pull request merged**